src/graph.rs

```diff
@@ -354,6 +302,78 @@ impl From<String> for Value {
     }
 }
 
+impl fmt::Display for Value {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            Value::Null => write!(f, "#null"),
+            Value::Boolean(value) => write!(f, "{}", value),
+            Value::Integer(value) => write!(f, "{}", value),
+            Value::String(value) => write!(f, "\"{}\"", value),
+            Value::List(values) => {
+                write!(f, "[")?;
+                for elem in values {
+                    write!(f, "{},", elem)?;
+                }
+                write!(f, "]")
+            }
+            Value::Set(values) => {
+                write!(f, "{{")?;
+                for elem in values {
+                    write!(f, "{},", elem)?;
+                }
+                write!(f, "}}")
+            }
+            Value::SyntaxNode(value) => <dyn fmt::Display>::fmt(value, f),
+            Value::GraphNode(value) => <dyn fmt::Display>::fmt(value, f),
```

Comment:
Compiler complained about "multiple applicable items in scope" when writing `value.fmt(f)` here. I couldn't figure out which items that were, but calling it like this fixed that. Alternatively, we could `write!("{}", value)`, but this would add another indirection.

Reply:
The other will be `Debug::fmt`.  The `write!` formulation _might_ get translated into an efficient call; I'm not quite sure how the macro is implemented.  I think you can simplify what you have to:

``` rust
fmt::Display::fmt(value, f)
```